ui/components/multichain/account-list-menu/account-list-menu.js

```diff
@@ -64,6 +66,34 @@ const ACTION_MODES = {
   IMPORT: 'import',
 };
 
+/**
+ * Merges ordered accounts with balances with each corresponding account data from internal accounts
+ *
+ * @param accountsWithBalances - ordered accounts with balances
+ * @param internalAccounts - internal accounts
+ * @returns merged accounts list with balances and internal account data
+ */
+const mergeAccounts = (accountsWithBalances, internalAccounts) => {
+  return accountsWithBalances.map((account) => {
+    const internalAccount = internalAccounts.find(
+      (intAccount) => intAccount.address === account.address,
+    );
+    if (internalAccount) {
+      return {
+        ...account,
+        ...internalAccount,
+        name: internalAccount.metadata.name || account.name,
+        keyring: internalAccount.metadata.keyring.type,
+        label: getAccountLabel(
+          internalAccount.metadata.keyring.type,
+          internalAccount,
+        ),
+      };
+    }
+    return account;
+  });
+};
+
```

Comment:
This change seems unrelated to the PR, where is it coming from?

Reply:
It is needed because we have not replaced MetaMask Identities with internal accounts
so when the component calls `const accounts = useSelector(getMetaMaskAccountsOrdered);` to get a list of accounts those accounts do not have the data that internalAccounts do BUT they have balances which internal accounts do not have. So we have to merge the results of that selector with the data we also need from the internal accounts in order to have all the data WITH the balances

Reply:
Ah ok, that will be a little bit of technical debt, lets remember to resolve with the rest of the accounts-controller changes